src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1154,24 +1157,43 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
-                                    ConflictStatus.AlreadyMerged)
+                                if (localAttachment == null)
                                 {
                                     database.DocumentsStorage.AttachmentsStorage.PutDirect(context, attachment.Key, attachmentName,
                                         contentType, attachment.Base64Hash, attachment.ChangeVector);
+                                    break;
                                 }
 
+                                var mergedChangedVector = attachment.ChangeVector;
+                                var status = ChangeVectorUtils.GetConflictStatus(attachment.ChangeVector, localAttachment.ChangeVector);
+                                switch (status)
+                                {
+                                    case ConflictStatus.Update:
+                                        database.DocumentsStorage.AttachmentsStorage.PutDirect(context, attachment.Key, attachmentName,
+                                            contentType, attachment.Base64Hash, mergedChangedVector);
+                                        break;
+                                    case ConflictStatus.Conflict:
+                                        mergedChangedVector = ChangeVectorUtils.MergeVectors(attachment.ChangeVector, localAttachment.ChangeVector);
+                                        goto case ConflictStatus.Update;
+                                    case ConflictStatus.AlreadyMerged:
+                                        break;
+                                    default:
+                                        throw new ArgumentOutOfRangeException();
+                                }
+                                
                                 break;
 
                             case AttachmentTombstoneReplicationItem attachmentTombstone:
-
                                 var tombstone = AttachmentsStorage.GetAttachmentTombstoneByKey(context, attachmentTombstone.Key);
                                 if (tombstone != null && ChangeVectorUtils.GetConflictStatus(item.ChangeVector, tombstone.ChangeVector) == ConflictStatus.AlreadyMerged)
                                     continue;
 
-                                database.DocumentsStorage.AttachmentsStorage.DeleteAttachmentDirect(context, attachmentTombstone.Key, false, "$fromReplication", null,
-                                    rcvdChangeVector,
-                                    attachmentTombstone.LastModifiedTicks);
+                                var hashSlice = database.DocumentsStorage.AttachmentsStorage.DeleteAttachmentDirect(context, attachmentTombstone.Key, isPartialKey: false,
+                                    "$fromReplication", expectedChangeVector: null, rcvdChangeVector, attachmentTombstone.LastModifiedTicks, deleteAttachmentStream: false);
+                                
+                                if(hashSlice.HasValue)
```

Comment:
Under what scenario is this _not_ true?

Reply:
when `DeleteAttachmentDirect` can't find the metadata we want to delete, the slice have default value

Reply:
Could be that the last batch item deleted a duplicate, and in the next batch item we get a tombstone for it